### PR TITLE
2155 api integration

### PR DESCRIPTION
**Pull request by GSuma21**

### CODE REVIEW

File: src/app/modules/layout/components/session-listing/session-listing.component.ts [hunk 1]
```diff
@@ -1,195 +1,58 @@
...
+    this.getAllSession();
+  }
 
+  onClickViewMore() {
+    this.lastIndex = this.cardDetails.length
   }
+  getAllSession() {
+    let config = {
+      url: API_CONSTANTS.HOME_SESSION + this?.page + '&limit=' + this?.limit,
+      payload: {}
```
Comment (vishwanath1004) [inline src/app/modules/layout/components/session-listing/session-listing.component.ts:48]: you can remove payload key if you are not passing anything in that.

File: src/app/modules/layout/components/session-listing/session-listing.component.ts [hunk 2]
```diff
@@ -1,195 +1,58 @@
...
+
   }
 
   ngOnInit(): void {
-
+    
     if (this.selectedPage == '/enrolled-sessions') {
       this.cardHeading = "MY_SESSIONS"
-      this.cardDetails = this.apiCardDetails.result.mySessions;
+      this.status = "enrolled-sessions";
```
Comment (vishwanath1004) [inline src/app/modules/layout/components/session-listing/session-listing.component.ts:34]: list all the session status in const file and get from there.

File: src/app/modules/layout/components/session-listing/session-listing.component.ts [hunk 3]
```diff
@@ -1,195 +1,58 @@
...
+  onClickViewMore() {
+    this.lastIndex = this.cardDetails.length
   }
+  getAllSession() {
+    let config = {
+      url: API_CONSTANTS.HOME_SESSION + this?.page + '&limit=' + this?.limit,
+      payload: {}
+    };
 
+    this.apiService.get(config).subscribe((data: any) => {
```
Comment (vishwanath1004) [inline src/app/modules/layout/components/session-listing/session-listing.component.ts:51]: for api's we have to add loader and also success and error messages. If added in api service thats fine.

File: src/app/modules/layout/layout.component.ts [hunk 4]
```diff
@@ -13,5 +13,6 @@ export class LayoutComponent implements OnInit {
   constructor() { }
 
   ngOnInit(): void {
+
```
Comment (vishwanath1004) [inline src/app/modules/layout/layout.component.ts:16]: Remove this extra line unnecessarily its showing one file change.

File: src/app/shared/components/session-card/session-card.component.ts [hunk 5]
```diff
@@ -8,9 +10,31 @@ import { Router } from '@angular/router';
...
   @Input() cardData: any;
-  constructor(private router: Router) { }
+  buttonConfig:any;
+  isCreator: boolean;
+  userData: any;
+  constructor(private localStorage:LocalStorageService,private translate: TranslateService) { }
 
-  ngOnInit(): void {
+  async ngOnInit() {
+    console.log(this.cardData)
```
Comment (vishwanath1004) [inline src/app/shared/components/session-card/session-card.component.ts:19]: remove console logs

File: src/app/modules/layout/components/session-listing/session-listing.component.ts (hunk 1 above)
Comment (kiranharidas187) [reply src/app/modules/layout/components/session-listing/session-listing.component.ts:48]: And this API call should go to a service

File: src/app/modules/auth/login/login.component.ts [hunk 6]
```diff
@@ -71,6 +73,11 @@ export class LoginComponent implements OnInit {
         this.localStorage.saveLocalData(localKeys.REMEMBER_ME, btoa(JSON.stringify(this.loginForm.myForm.value)) )
       } 
       this.router.navigate(['/home']);
+      this.toastService.showMessage("You have loggedin successfully",'success');
```
Comment (kiranharidas187) [inline src/app/modules/auth/login/login.component.ts:76]: translate

**Pull request merged**